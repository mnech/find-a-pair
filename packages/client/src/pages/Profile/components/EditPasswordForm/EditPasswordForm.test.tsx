import React from 'react';
import { render } from '@testing-library/react';
import { EditPasswordForm } from './index';

describe('EditPasswordForm', () => {
  describe('should be rendered correctly EditPasswordForm component', () => {
    test('default', () => {
      let isEditPassword = false;
      const setIsEditPassword = (value: boolean) => (isEditPassword = value);

      const { asFragment } = render(
        <EditPasswordForm setIsEditPassword={setIsEditPassword} />,
      );
      expect(asFragment()).toMatchInlineSnapshot(`
        <DocumentFragment>
          <form
            class="form-edit-password"
            novalidate=""
          >
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="oldPassword"
              >
                Старый пароль
              </label>
              <input
                class="form-control is-invalid"
                id="oldPassword"
                name="oldPassword"
                placeholder="Введите старый пароль"
                required=""
                type="password"
                value=""
              />
              <div
                class="invalid-feedback"
              >
                Введите верный старый пароль
              </div>
            </div>
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="newPassword"
              >
                Новый пароль
              </label>
              <input
                class="form-control is-invalid"
                id="newPassword"
                name="newPassword"
                placeholder="Введите новый пароль"
                required=""
                type="password"
                value=""
              />
              <div
                class="invalid-feedback"
              >
                Введите верный новый пароль
              </div>
            </div>
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="repeatNewPassword"
              >
                Повторите новый пароль
              </label>
              <input
                class="form-control is-invalid"
                id="repeatNewPassword"
                name="repeatNewPassword"
                placeholder="Введите повторно новый пароль"
                required=""
                type="password"
                value=""
              />
              <div
                class="invalid-feedback"
              >
                Пароли не совпадают
              </div>
            </div>
            <button
              class="button btn btn-primary"
              type="submit"
            >
              Submit
            </button>
          </form>
        </DocumentFragment>
      `);
    });
  });
});
