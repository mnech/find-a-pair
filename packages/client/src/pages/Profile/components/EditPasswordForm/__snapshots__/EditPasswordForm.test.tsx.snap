// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPasswordForm should be rendered correctly EditPasswordForm component default 1`] = `
<DocumentFragment>
  <form
    class="form-edit-password"
    novalidate=""
  >
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="oldPassword"
      >
        Старый пароль
      </label>
      <input
        class="form-control is-invalid"
        id="oldPassword"
        name="oldPassword"
        placeholder="Введите старый пароль"
        required=""
        type="password"
        value=""
      />
      <div
        class="invalid-feedback"
      >
        Введите верный старый пароль
      </div>
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="newPassword"
      >
        Новый пароль
      </label>
      <input
        class="form-control is-invalid"
        id="newPassword"
        name="newPassword"
        placeholder="Введите новый пароль"
        required=""
        type="password"
        value=""
      />
      <div
        class="invalid-feedback"
      >
        Введите верный новый пароль
      </div>
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="repeatNewPassword"
      >
        Повторите новый пароль
      </label>
      <input
        class="form-control is-invalid"
        id="repeatNewPassword"
        name="repeatNewPassword"
        placeholder="Введите повторно новый пароль"
        required=""
        type="password"
        value=""
      />
      <div
        class="invalid-feedback"
      >
        Пароли не совпадают
      </div>
    </div>
    <button
      class="button btn btn-primary"
      type="submit"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
